From fbaa3be62063ea0a4cf733d610abc00ed6a510ae Mon Sep 17 00:00:00 2001
From: Nathaniel Russell <46272571+nater1983@users.noreply.github.com>
Date: Wed, 25 Oct 2023 03:51:02 +0000
Subject: [PATCH 1/3] Update meson.build

Add subdir('xdg-portal')
Install a portals.conf file for the pantheon session
---
 meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/meson.build b/meson.build
index 2d09a4f..551d0de 100644
--- a/meson.build
+++ b/meson.build
@@ -46,3 +46,4 @@ conf_data.set_quoted('VERSION', meson.project_version())
 subdir('data')
 subdir('po')
 subdir('src')
+subdir('xdg-portal')

From 5d666395addfd1d9b44b15e1d484c94b15d6ae49 Mon Sep 17 00:00:00 2001
From: Nathaniel Russell <46272571+nater1983@users.noreply.github.com>
Date: Wed, 25 Oct 2023 03:53:05 +0000
Subject: [PATCH 2/3] Create pantheon-portals.conf

---
 xdg-portal/pantheon-portals.conf | 3 +++
 1 file changed, 3 insertions(+)
 create mode 100644 xdg-portal/pantheon-portals.conf

diff --git a/xdg-portal/pantheon-portals.conf b/xdg-portal/pantheon-portals.conf
new file mode 100644
--- /dev/null
+++ b/xdg-portal/pantheon-portals.conf
@@ -0,0 +1,5 @@
+[preferred]
+default=pantheon;gtk;
+org.freedesktop.impl.portal.FileChooser=io.elementary.files;
+org.freedesktop.impl.portal.Secret=gnome-keyring;
+org.freedesktop.impl.portal.Settings=io.elementary.settings-daemon;gtk;*;

From 7bdc55e66fa4d2c08360cfcefaf8fb02011c66e8 Mon Sep 17 00:00:00 2001
From: Nathaniel Russell <46272571+nater1983@users.noreply.github.com>
Date: Wed, 25 Oct 2023 03:53:34 +0000
Subject: [PATCH 3/3] Create meson.build

---
 xdg-portal/meson.build | 4 ++++
 1 file changed, 4 insertions(+)
 create mode 100644 xdg-portal/meson.build

diff --git a/xdg-portal/meson.build b/xdg-portal/meson.build
new file mode 100644
index 0000000..bf6cdb8
--- /dev/null
+++ b/xdg-portal/meson.build
@@ -0,0 +1,4 @@
+install_data(
+  'pantheon-portals.conf',
+  install_dir: join_paths(get_option('datadir'), 'xdg-desktop-portal'),
+)
